src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx

```diff
@@ -96,12 +92,13 @@ function QueryEditor(props: QueryEditorProps) {
         type,
         theme,
         changeUserInput,
+        setQueryResult,
         showPreview,
     } = props;
     const {tenantPath: savedPath} = executeQuery;
 
-    const [resultType, setResultType] = React.useState<ValueOf<typeof RESULT_TYPES>>();
-    const [isResultLoaded, setIsResultLoaded] = React.useState(false);
+    const isResultLoaded = Boolean(executeQuery.result);
```

Comment:
why only `executeQuery.result`?

Reply:
executeQuery.result can be both execute or explain type